recipes/geant4/run_test.sh

```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+cmake ${PREFIX}/share/Geant4-${PKG_VERSION}/examples/basic/B1
+make
+source ${PREFIX}/bin/geant4.sh
```

Comment:
The contents of `geant4.(c)sh` should be moved into a post setup hook as is done for ROOT (see [here](https://github.com/conda-forge/root-feedstock/blob/980d1fe10dfd4f365846b9ccc44295d5757f3a99/recipe/build.sh#L147-L156)) and the setup scripts should then be removed as they do harmful things like set `LD_LIBARY_PATH`.

Reply:
Working on it

Reply:
Actually these scripts are merely setting some environment variables with the locations of the Geant4 data files. Nothing is done to `LD_LIBRARY_PATH` or `PATH` AFAICS.

I could move the contents to the activate/deactivate files but that may cause confusion among the Geant4 users that expect `geant4.(c)sh` to be present.

Reply:
Ah, the [LCG](https://ep-dep-sft.web.cern.ch/document/lcg-releases) build I looked at has them differently.

I still find it nicer to have the environment variables setup automatically rather than relying on on users remembering to source `geant4.sh`. It can also unset the variables when changing conda environments. I also added dummy `geant4.(c)sh` scripts to ensure they're not picked up from from a different GEANT4 installation. What do you think?

Reply:
These activation scripts won't be invoked in the root environment right? So there won't be an easy way for the user to get these variables set up then?

I don't think there's anything wrong with the user relying on these scripts since this is what the Geant4 docs tell people to use.

Reply:
If a "recent" conda installation is used they will be set automatically .`conda` is now a function sourced from a script (`$CONDA_DIR/etc/profile.d/conda.sh`) rather than requiring you to editing your PATH. This means it doesn't affect anything until you run `conda activate` and `conda install` or `conda activate` can transparently modify environment variables.

> So there won't be an easy way for the user to get these variables set up then?

`conda activate` or `conda activate root` should work on newer installations and `source activate` or `souce activate root` should work on any installation. Some other conda packages won't work without the environment being set up correctly (most notably the compilers) and I've patched the `geant4.(c)sh` scripts to print a message telling users what to do.

Does this seem reasonable? If you disagree, you can remove the last 5 lines of the build script.

I know I'm a little idealistic about this but I really don't think people should ever have to source magic scripts to make packages work correctly, once the environment is active everything should just work. Personally I would prefer to include the default paths inside the binaries and for the environment variables to be used only for overriding them but that's a suggestion for another day...

Reply:
While the decision of the Geant4 devs to expect the user to source a shell script to get the library working is certainly questionable, it is one we have to live with.

Therefore I believe that sourcing the shell script should always result in a working library, and this should be independent of whether the user activated the root environment or not.

I am therefore in favor of keeping the Geant4 scripts as generated by cmake, and dispose of the activation scripts.

Reply:
Why not keep both?

Reply:
I think one of the reasons that the Geant4 developers introduced this shell script is to allow users to override the default datasets.

If you're going to rely on activation scripts to set the environment variables, then things may get messy quickly as the value of these variables will then depend on the order of the scripts that get sourced: the ones provided by the activation script or the script with the custom variables.

Reply:
Hi Chris... any final thoughts on this?

Reply:
Sorry I missed your last reply and I hope I'm not being too awkward about this, I just want GEANT4 to be packaged as nicely as possible as it's a dependency of some other packages I will be adding.

I think it's more likely to get messy if the activation scripts don't set these variables. If someone runs `source geant4.sh` then `conda activate my-environment` they will get these variables from a different GEANT4 installation which probably isn't what they want.

Also I just noticed `geant4.(c)sh` does change the `PATH` and `LD_LIBRARY_PATH`[1] on Linux and this will break things in unexpected ways.

How about some other ideas (which aren't mutually exclusive):

1. The activation script print a warning if any of the variables are already set.
```bash
$ export G4NEUTRONHPDATA=/my/custom/data/directory
$ conda activate my-environment
WARNING: $G4NEUTRONHPDATA has been changed from "/my/custom/data/directory" to "/conda/share/Geant4/data/G4NDL4.5"
$ 
```
2. Create a package called `geant4-base` which doesn't include the data at all
    * There is then metapackage called `geant4` which depends on `geant4-base`,  `geant4-data-g4ndl` and other `geant4-data-.*` packages which are pinned to the default versions for that geant4 release but users can still install different versions of the data if they want.
    * The `$G4.*DATA` variables are then set by each of the data packages.
    * Actually I really like this idea as as there can be four variants of  `geant4`:
        * `default`: No QT support and pinned to the exact data package versions
        * `custom_data`: No QT support and any version of the data packages can be used
        * `qt-default`: QT support and pinned to the exact data package versions
        * `qt-custom_data`: QT support and any version of the data packages can be used
    * Running `conda install geant4-data-g4ndl=4.3` will automatically switch to the `custom_data` variant as version 4.5 is the default.
    * Downstream packages depend on `geant4` which will can be any of the four variants. This is way that `numpy` has support for both `openblas` and `mkl`.

Having thought about it I'm inclined to implement both ideas, what do you think?

[1] Search for "Contents of $PREFIX/bin/geant4.sh is" in [this log](https://circleci.com/gh/conda-forge/staged-recipes/36812?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link).

Reply:
Sure sounds good but looks like a lot of work.